### DISCUSSION TITLE
How to create a config? What do the parameters mean?

### DISCUSSION THREAD

**Discussion by Troodi** [swar/Swar-Chia-Plot-Manager#210] [Q&A] [answered]
What is max_concurrent is the number of rafts that will be sequentially written within one task?
How do I record 5 plots in parallel? Jobs are responsible for sequential write plots only? What does max_concurrent_with_start_early mean? Can anyone show the config file with a real example with 2 parallels?

**Comment by swar** [comment, marked as answer]
I think the README is pretty thorough with explaining some of these things, but so many people have had a conversation about this already.

**Reply by EclipzzeX** [reply]
This is a great YT video on setting up PlotManager.  He did this all for a windows based miner that I was able to translate to 'nix, and get up and running.  https://www.youtube.com/watch?v=Wl6FVRAj86I

**Reply by Cminer3207** [reply]
He doesn't know how to do multiple Drives. 
He also forgot to tell you to set venv before installing Python. 
Better to skip this guy

**Reply by Wilkugmo** [reply]
Could you please explain or link a guide to what "set venv before installing Python" means?
I (roughly) followed his guide to set up swar manager and everything works fine but you got me worried.

**Comment by Cminer3207** [comment]
Python has many possible configuration settings. 
To isolate one configuration of Python (the one being used by SWAR) from others... you install it inside it’s own isolation environment. 
Thisnis called a virtual environment.

**Comment by Cminer3207** [comment]
After you extract SWAR and place it in a folder...
And after you install Python on your computer...

You will create a virtual environment for your Swar’s Python module...

1. open PowerShell (run as Administrator) and then navigate into the SWAR folder using 
 cd C:\path to SWAR folder
When you are in the folder, you will create a Virtual Environment in that folder by typing:
python -m venv virt

“virt” is the name of your virtual environment. 
You will know it’s there because you will see a folder named “virt”.

Then you will ‘activate’ this virtual environment by typing the following:

virt\Scripts\activate

Then, once this environment is installed and activated, you will install Python for your SWAR  by typing:
pip install -r requirements.txt

**Reply by Cminer3207** [reply]
Just go slow and follow the SWAR install steps exactly.

**Reply by Cminer3207** [reply]
Some problems you might encounter:
“Scripts not allowed” 
By default Windows disables scripts because they can do damage. 
Open PowerShell and enter:

set-executionpolicy remotesigned

“Mapping Not allowed.”
You left some spaces in your config.YAML file. There must be one space, and one space only, between the parameter and setting. 
example: 
location: C:\Pathway   CORRECT
location:C:\Pathway    FAIL

**Reply by Wilkugmo** [reply]
If I understand correctly everything worked fine for me cause i just downloaded python and swar manager is my only configuration setting. But do you know how and if I should change it now when everything is set up and swar manager is plotting?

**Reply by Wilkugmo** [reply]
*I turned scripts on long time ago for a different reason and right now Swar runs smoothly.

**Reply by Cminer3207** [reply]
I wouldn’t... if it works it works... don’t mess up a good thing. 
Some people will have difficulty without a venv because we use Python for other things....

**Reply by Wilkugmo** [reply]
That makes sense, I was planning to learn how to code in python tho :D
Then I would **have to** turn the plotting off and follow your instructions* before using other configuration settings right?



*open PowerShell (run as Administrator) and then navigate into the SWAR folder using
cd C:\path to SWAR folder
When you are in the folder, you will create a Virtual Environment in that folder by typing:
python -m venv virt
“virt” is the name of your virtual environment.
You will know it’s there because you will see a folder named “virt”.

Then you will ‘activate’ this virtual environment by typing the following:

virt\Scripts\activate

**Reply by Cminer3207** [reply]
I made a copy of my YAMl. 
then deleted the old Swar folder. 
Extracted a new one from the zip.
Navigated to folder with Powershell.
installed the venv. 
installed requirements.txt
Then moved my YAMl back to the new folder and started again. 

Took about 2 minutes.

**Reply by Wilkugmo** [reply]
Will do that then, thank you for help.

**Comment by Cminer3207** [comment]
Without mining pools, we are all competing against each other. 

Helping you hurts me... 

We need pools.

**Reply by Wilkugmo** [reply]
I feel you, thanks for helping :D 
Hopefully the pool protocol will be up in a few days, so it's a ~week till pools are up.

**Reply by Wilkugmo** [reply]
Also, my few plots more or less per day won't matter anyway. It's the whales making hundreds of plots/day.

**Reply by Cminer3207** [reply]
Lol. 
My comment was philosophical...
Mining pools are a stronger community because we all work together.

**Reply by Troodi** [reply]
But we need replot? Old plots not will not be work

**Reply by Wilkugmo** [reply]
Yes, we do. You can fill what you have left once pool ready plots are up and then replace old plots.
Maybe you win chia lottery.

**Reply by Cminer3207** [reply]
Who knows what pool will require. 
I’d think the developers could adding pooling so we can keep existing plots. 
Or maybe we all transition to Plotting As A Service... and sell the plots themselves, with their current keys, to the Pool Managers for a flat fee.

**Reply by Troodi** [reply]
It would be cool
